### PR TITLE
Metric learning ce

### PR DESCRIPTION
**Pull request by zhengya01**

### CODE REVIEW

**Review by kolinwei — APPROVED** [review]

File: fluid/PaddleCV/metric_learning/imgtool.py [hunk 1]
```diff
@@ -35,14 +42,18 @@ def random_crop(img, size, scale=None, ratio=None):
...
+        target_area = img.shape[0] * img.shape[1] * (scale_min + scale_max) / 2.
     target_size = math.sqrt(target_area)
     w = int(target_size * w)
     h = int(target_size * h)
 
     i = random.randint(0, img.shape[0] - h)
     j = random.randint(0, img.shape[1] - w)
+    if is_ce:
+        i = int(img.shape[0] - h) // 2
+        j = int(img.shape[1] - w) // 2
```
Comment (qingqing01) [inline fluid/PaddleCV/metric_learning/imgtool.py]: You also can set rand seed for numpy, please refer image classification. So do not add `ce` branch in reader.

File: fluid/PaddleCV/human_pose_estimation/lib/coco_reader.py [hunk 2]
```diff
@@ -31,6 +31,10 @@
 from lib.base_reader import visualize, generate_target
 from pycocotools.coco import COCO
 
+is_ce = int(os.environ.get('is_ce', 0))
+if is_ce:
+    random.seed(0)
```
Comment (qingqing01) [inline fluid/PaddleCV/human_pose_estimation/lib/coco_reader.py:36]: 这个seed设置，放到`train.py`line102行之后就行吧？

File: fluid/PaddleCV/human_pose_estimation/train.py [hunk 3]
```diff
@@ -143,15 +153,25 @@ def if_exist(var):
         use_cuda=True if args.use_gpu else False, loss_name=loss.name)
     fetch_list = [image.name, loss.name, output.name]
 
+    total_time = 0
+    last_loss = 0
     for pass_id in range(params["num_epochs"]):
         for batch_id, data in enumerate(train_reader()):
+            if args.enable_ce and args.batch_num is not None:
+                if batch_id >= args.batch_num:
```
Comment (qingqing01) [inline fluid/PaddleCV/human_pose_estimation/train.py:161]: 这里是CE时一个epoc，只训练batch_num个？

File: fluid/PaddleCV/metric_learning/train_elem.py [hunk 4]
```diff
@@ -230,10 +231,16 @@ def if_exist(var):
             train_info = [0, 0, 0, 0]
 
         totalruntime += period
+        total_time += 1
         
-        if iter_no % args.test_iter_step == 0 and iter_no != 0:
+        #if iter_no % args.test_iter_step == 0 and iter_no != 0:
```
Comment (qingqing01) [inline fluid/PaddleCV/metric_learning/train_elem.py]: Remove this line.

File: fluid/PaddleCV/metric_learning/train_elem.py [hunk 5]
```diff
@@ -230,10 +231,16 @@ def if_exist(var):
...
+        if (iter_no % args.test_iter_step == 0 and iter_no != 0) or args.enable_ce:
             f, l = [], []
             for batch_id, data in enumerate(test_reader()):
+                if args.enable_ce:
+                    if batch_id > 1:
+                        break
```
Comment (qingqing01) [inline fluid/PaddleCV/metric_learning/train_elem.py]: 这里是只跑一个batch？ 能减少数据量，而不是改代码？
